src/System.Runtime/tests/System/StringTests.cs

```diff
@@ -1653,6 +1653,51 @@ public static void Join_ObjectArray_Null_ThrowsArgumentNullException()
             Assert.Throws<ArgumentNullException>("values", () => string.Join("--", (IEnumerable<object>)null));
         }
 
+        public static object[][] Join_Char_TestData() => new object[][]
+        {
+            new object[] { '$', new object[] { }, 0, 0, "" },
+
+            new object[] { '$', new object[] { null }, 0, 1, "" },
+            new object[] { '$', new object[] { "" }, 0, 1, "" },
+            new object[] { '$', new object[] { "Foo" }, 0, 1, "Foo" },
+
+            new object[] { '$', new object[] { null, null, null }, 0, 3, "$$$$" },
+            new object[] { '$', new object[] { "", "", "" }, 0, 3, "$$$$" },
+
+            new object[] { '$', new object[] { null, "Bar", null }, 0, 3, "$Bar$" },
+            new object[] { '$', new object[] { "", "Bar", "", }, 0, 3, "$Bar$" },
+            new object[] { '$', new object[] { "Foo", "Bar", "Baz" }, 0, 3, "Foo$Bar$Baz" },
+
+            new object[] { '$', new object[] { "Foo", "Bar", "Baz" }, 0, 3, "" },
+            new object[] { '$', new object[] { "Foo", "Bar", "Baz" }, 1, 1, "Bar" },
+        };
+
+        [Theory]
+        [MemberData(nameof(Join_Char_TestData))]
+        public static void Join_Char_Test(char separator, object[] values, int startIndex, int count, string expected)
```

Comment:
@justinvp I tried make so. But separated file with conditional compilation was not compiled. I am think how correct include conditional compilation.